pygdf/categorical.py

```diff
@@ -155,6 +155,52 @@ def to_pandas(self, index=None):
                                          ordered=self._ordered)
         return pd.Series(data, index=index)
 
+    def to_pandas(self, index=None):
+        return pd.Series(self.to_array(fillna='pandas'), index=index)
+
+    def _unique_segments(self):
+        """ Common code for unique, unique_count and value_counts"""
+        # make dense column
+        densecol = self.replace(data=self.to_dense_buffer(), mask=None)
+        # sort the column
+        sortcol, _ = densecol.sort_by_values(ascending=True)
+        # find segments
+        sortedvals = sortcol.to_gpu_array()
+        segs, begins = cudautils.find_segments(sortedvals)
+        return segs, sortedvals
+
+    def unique(self, type='sort'):
+        # type variable will indicate what algorithm to use to
+        # calculate unique, not used right now
+        if type is not 'sort':
+            msg = 'non sort based unique() not implemented yet'
+            raise NotImplementedError(msg)
+        segs, sortedvals = self._unique_segments()
+        # gather result
+        out = cudautils.gather(data=sortedvals, index=segs)
+        return self.replace(data=Buffer(out), mask=None)
```

Comment:
Correct me if I'm wrong, but we should utilize the dictionary here instead of scanning the today.